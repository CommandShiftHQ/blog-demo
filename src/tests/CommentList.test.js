import React from "react";
import { render } from "@testing-library/react";
import CommentList from "../components/CommentList";

const comments = [
  {
    body: "Test comment body 1",
    username: "Test comment username 1",
    uuid: `Test id 123`,
  },
  {
    body: "Test comment body 2",
    username: "Test comment username 2",
    uuid: "Test id 124",
  }
];

describe("CommentList", () => {
  it("renders as expected", () => {
    const { asFragment } = render(
      <CommentList comments={comments} />
    );
    const component = asFragment();
    expect(component).toMatchSnapshot();
  });

  it("render expected number of comments", () => {
    const { getAllByTestId } = render(
      <CommentList comments={comments} />
    );

    expect(getAllByTestId("comment")).toHaveLength(2);
  });
});