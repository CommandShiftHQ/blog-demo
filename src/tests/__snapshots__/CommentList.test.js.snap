// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentList renders as expected 1`] = `
<DocumentFragment>
  <ul>
    <li
      data-testid="comment"
    >
      <p>
        <strong>
          Test comment username 1: 
        </strong>
        Test comment body 1
      </p>
    </li>
    <li
      data-testid="comment"
    >
      <p>
        <strong>
          Test comment username 2: 
        </strong>
        Test comment body 2
      </p>
    </li>
  </ul>
</DocumentFragment>
`;
